Source/Serilog.Exceptions.Refit/Destructurers/ApiExceptionDestructurer.cs

```diff
@@ -0,0 +1,61 @@
+namespace Serilog.Exceptions.Refit.Destructurers
+{
+    using System;
+    using System.Collections.Generic;
+    using global::Refit;
+    using Serilog.Exceptions.Core;
+    using Serilog.Exceptions.Destructurers;
+
+    /// <summary>
+    /// A destructurer for the Refit <see cref="ApiException"/>.
+    /// </summary>
+    /// <seealso cref="ExceptionDestructurer" />
+    public class ApiExceptionDestructurer : ExceptionDestructurer
+    {
+        private readonly bool destructureCommonExceptionProperties = true;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApiExceptionDestructurer"/> class.
+        /// </summary>
+        /// <param name="destructureCommonExceptionProperties">Destructure common public Exception properties or not.</param>
+        public ApiExceptionDestructurer(bool destructureCommonExceptionProperties = true) => this.destructureCommonExceptionProperties = destructureCommonExceptionProperties;
+
+        /// <inheritdoc cref="IExceptionDestructurer.TargetTypes"/>
+        public override Type[] TargetTypes => new[] { typeof(ApiException) };
+
+        /// <inheritdoc />
+        public override void Destructure(Exception exception, IExceptionPropertiesBag propertiesBag, Func<Exception, IReadOnlyDictionary<string, object?>?> destructureException)
+        {
+            if (this.destructureCommonExceptionProperties)
+            {
+                base.Destructure(exception, propertiesBag, destructureException);
+            }
+            else
+            {
+                // Argument checks are usually done in <see cref="ExceptionDestructurer.Destructure"/>
+                // but as we didn't call this method we need to do the checks here.
+                if (exception is null)
+                {
+                    throw new ArgumentNullException(nameof(propertiesBag));
+                }
+
+                if (propertiesBag is null)
+                {
+                    throw new ArgumentNullException(nameof(propertiesBag));
+                }
+
+                if (destructureException is null)
+                {
+                    throw new ArgumentNullException(nameof(destructureException));
+                }
+            }
+
+#pragma warning disable CA1062 // Validate arguments of public methods
+            var apiException = (ApiException)exception;
+            propertiesBag.AddProperty(nameof(ApiException.Uri), apiException.Uri);
+            propertiesBag.AddProperty(nameof(ApiException.Content), apiException.Content);
```

Comment:
I've not used Refit before. Does `ApiException.Content` effectively contain the response body? That could potentially be very large and maybe something you wouldn't want to log by default.

Reply:
Yes this is the HTTP response body. Yes I hesitated on this property, this could potentially be very large. I could add an option in the constructor `bool destructureContent` with a default of false ?

Reply:
Probably content should be destructured, that's fine. Maybe we should just limit the payload to some reasonable limit (like 2kb) and let user configure otherwise.

Reply:
All logging frameworks used by ASP.NET omit the content. We should do the same by default and provide an option to enable it.

Reply:
I made it optional (default disabled). Let me know if this is ok and I will add some tests for it.